counsel.el

```diff
@@ -3255,6 +3255,8 @@ All blank strings are deleted from `kill-ring' by default."
   "Return list of kills for `counsel-yank-pop' to complete.
 Returned elements satisfy `counsel-yank-pop-filter' and are
 unique under `equal-including-properties'."
+  ;; Refresh kill-ring in the presence of `interprogram-paste-function'
+  (current-kill 0)
```

Comment:
I'm not convinced it's a good idea to unconditionally allow `kill-ring` to be prepended to every time `counsel--yank-pop-kills` is called.

One of the problems is that `current-kill` throws an error if `kill-ring` is empty and `interprogram-paste-function` is or returns `nil`.

Another problem is that other callers, such as `counsel-yank-pop-action-rotate`, do not take this behaviour of `counsel--yank-pop-kills` into account.

Yet another problem is that, with vanilla `yank`/`yank-pop`, interprogram paste(s) are only prepended to `kill-ring` when: 
* `yank`, `yank-pop`, or `rotate-yank-pointer` is called with a zero prefix argument; or
* `save-interprogram-paste-before-kill` is non-`nil` and a kill is performed.

So, before continuing, we must decide whether to:
1. honour the vanilla Emacs behaviour of only prepending to `kill-ring` in certain cases, as described above; or
2. always prepend to `kill-ring` before any `counsel-yank-pop*` action.

I am in favour of (2), but I wanted to make sure that everyone is aware of and agrees with this divergence from vanilla Emacs behaviour.